File: project/jimmer-core/src/test/java/org/babyfish/jimmer/ComplexTest.java
package org.babyfish.jimmer;

import org.babyfish.jimmer.model.Complex;
import org.babyfish.jimmer.model.ComplexDraft;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ComplexTest {

    @Test
    public void test() {
        Complex a = ComplexDraft.$.produce(c -> {
            c.setReal(5);
            c.setImage(7);
        });
        Complex b = ComplexDraft.$.produce(a, c -> {
           c.setReal(c.getReal() - 1);
           c.setImage(c.getImage() - 1);
        });
        Assertions.assertEquals(
                "{\"real\":5.0,\"image\":7.0}",
                a.toString()
        );
        Assertions.assertEquals(
                "{\"real\":4.0,\"image\":6.0}",
                b.toString()
        );
    }
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/ComplexTest.kt
package org.babyfish.jimmer.kt

import org.babyfish.jimmer.kt.model.Complex
import org.babyfish.jimmer.kt.model.by
import kotlin.test.Test
import kotlin.test.expect

class ComplexTest {

    @Test
    fun test() {
        val complex = new(Complex::class).by {
            real = 5.0
            image = 7.0
        }
        val newComplex = new(Complex::class).by(complex) {
            real--
            image--
        }
        expect("""{"real":5.0,"image":7.0}""") { complex.toString() }
        expect("""{"real":4.0,"image":6.0}""") { newComplex.toString() }
    }
}